src-self-hosted/translate_c.zig

```diff
@@ -3095,31 +3121,53 @@ fn transCreateCompoundAssign(
+
+    if ((is_mod or is_div) and is_signed) {
+        const op_token = try appendToken(rp.c, .Equal, "=");
+        const op_node = try rp.c.a().create(ast.Node.InfixOp);
+        const builtin = if (is_mod) "@rem" else "@divTrunc";
+        const builtin_node = try transCreateNodeBuiltinFnCall(rp.c, builtin);
+        try builtin_node.params.push(try transCreateNodePtrDeref(rp.c, lhs_node));
         _ = try appendToken(rp.c, .Comma, ",");
-        try cast_node.params.push(rhs_node);
-        cast_node.rparen_token = try appendToken(rp.c, .RParen, ")");
-        rhs_node = &cast_node.base;
-    }
-    const rhs_bin = try transCreateNodeInfixOp(rp, scope, ref_node, bin_op, bin_token, rhs_node, .used, false);
+        try builtin_node.params.push(try transExpr(rp, scope, rhs, .used, .r_value));
+        builtin_node.rparen_token = try appendToken(rp.c, .RParen, ")");
+        _ = try appendToken(rp.c, .Semicolon, ";");
+        op_node.* = .{
+            .op_token = op_token,
+            .lhs = ref_node,
+            .op = .Assign,
+            .rhs = &builtin_node.base,
+        };
+        _ = try appendToken(rp.c, .Semicolon, ";");
+        try block_scope.block_node.statements.push(&op_node.base);
+    } else {
+        const bin_token = try appendToken(rp.c, bin_tok_id, bin_bytes);
+        var rhs_node = try transExpr(rp, scope, rhs, .used, .r_value);
 
-    _ = try appendToken(rp.c, .Semicolon, ";");
+        if (is_shift) {
+            const cast_node = try transCreateNodeBuiltinFnCall(rp.c, "@intCast");
+            const rhs_type = try qualTypeToLog2IntRef(rp, getExprQualType(rp.c, rhs), loc);
+            try cast_node.params.push(rhs_type);
+            _ = try appendToken(rp.c, .Comma, ",");
+            try cast_node.params.push(rhs_node);
+            cast_node.rparen_token = try appendToken(rp.c, .RParen, ")");
+            rhs_node = &cast_node.base;
+        }
 
-    const eq_token = try appendToken(rp.c, .Equal, "=");
-    const assign = try transCreateNodeInfixOp(rp, scope, ref_node, .Assign, eq_token, rhs_bin, .used, false);
-    try block_scope.block_node.statements.push(assign);
+        const rhs_bin = try transCreateNodeInfixOp(rp, scope, ref_node, bin_op, bin_token, rhs_node, .used, false);
+        _ = try appendToken(rp.c, .Semicolon, ";");
+
+        const eq_token = try appendToken(rp.c, .Equal, "=");
+        const assign = try transCreateNodeInfixOp(rp, scope, ref_node, .Assign, eq_token, rhs_bin, .used, false);
+        try block_scope.block_node.statements.push(assign);
+    }
 
     const break_node = try transCreateNodeBreak(rp.c, block_scope.label);
     break_node.rhs = ref_node;
     try block_scope.block_node.statements.push(&break_node.base);
     block_scope.block_node.rbrace = try appendToken(rp.c, .RBrace, "}");
     // semicolon must immediately follow rbrace because it is the last token in a block
-    _ = try appendToken(rp.c, .Semicolon, ";");
+    _ = try appendToken(rp.c, .Semicolon, ";4");
```

Comment:
This semicolon can be removed since it is not needed inside a grouped expression. Otherwise LGTM.

Reply:
Oops I forgot to remove my testing 😅 I'll get that fixed up.